src/RequestFactory.php

```diff
@@ -44,8 +44,20 @@ public static function setFakerResolver(Closure $resolver): void
         self::$fakerResolver = $resolver;
     }
 
-    public static function new(array $attributes = []): static
+    /**
+     * @param array<string, mixed>|static $attributes
+     */
+    public static function new(array|self $attributes = []): static
     {
+        /**
+         * When working with datasets, you may have a mixture of plain arrays and
+         * factories to work with depending on the complexity of the test. Here
+         * we avoid having to conditionally check for an array or factory.
+         */
+        if ($attributes instanceof static) {
+            return $attributes;
```

Comment:
I think we should create a new instance with its data instead of returning it self. 

If we think about inheritance, you could have an CreateAccountRequest and then sub classes for each account type. 

I could see a data provider, providing the inherited class and you would then expect to actually get the sub classes back 👀

Reply:
Good call. Updated